src/test/blockfilter_tests.cpp

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2018 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <blockfilter.h>
+#include <random.h>
+
+#include <boost/test/unit_test.hpp>
+
+BOOST_AUTO_TEST_SUITE(blockfilter_tests)
+
+BOOST_AUTO_TEST_CASE(gcsfilter_test)
+{
+    std::set<GCSFilter::Element> included_elements, excluded_elements;
+    for (int i = 0; i < 100; ++i) {
+        GCSFilter::Element rand1(32);
+        GetRandBytes(rand1.data(), rand1.size());
+        included_elements.insert(std::move(rand1));
+
+        GCSFilter::Element rand2(32);
+        GetRandBytes(rand2.data(), rand2.size());
```

Comment:
I recommend avoiding actual random values for unit tests, as generating random data is expensive and the tests may end up failing only sporadically, making it harder to debug (run the test 10 times and see the error once kinda thing).

A simple approach is to use a random generator "manually" once to generate 200 random numbers as a big static vector. Another is to do e.g. `srand(12345);` once and then use `rand()`, but I'm not sure if that has downsides (or if it's slow, too).